src/django/api/serializers/facility/facility_create_claim_serializer.py

```
@@ -87,6 +88,53 @@ def validate_date_range(opening_date, closing_date):
+    Parse an ISO-like date string to a date, ensuring it is not in the future.
+    Returns None for empty values.
+    """
+    if value in (None, ''):
+        return None
+
+    try:
+        parsed_date = date_parser.parse(value).date()
+    except (ValueError, OverflowError, TypeError):
+        raise DRFValidationError({
+            field_name: 'Please enter a valid date (not in the future).'
+        })
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)